__tests__/app.test.js

```diff
@@ -187,5 +187,45 @@ describe('GET article by id', () => {
                 })
                 })
                 })
+                describe('POST article comments', () => {
+                    test('returns status 400 when given invalid id ', () => {
+                     return request(app)
+                     .post('/api/articles/test/comments')
+                     .expect(400)
+                     .then(({ body }) => {
+                         expect(body.msg).toBe('Bad Request');
+                     })
+                    })
+                    test('returns status 404 when the given id is not found ', () => {
+                        return request(app)
+                        .post('/api/articles/99/comments')
+                        .expect(404)
+                       })
+                       test('returns status 404 when the given username is not found ', () => {
+                        return request(app)
+                        .post('/api/articles/99/comments')
+                        .send({username : 'test_name', body : 'Body of sample text?!! 1 + 2'})
+                        .expect(404)
+                       })
+                 test('returns the desired status code and array object properties ', () => {
+                 return request(app)
+                 .post('/api/articles/3/comments')
+                 .send({username : 'butter_bridge', body : 'Body of sample text?!! 1 + 2'})
+                 .expect(201)
+                 .then(({ body: {comment} }) => {
+                     expect(comment).toBeInstanceOf(Array)
+                     expect(comment.length).toBe(1)
+                     comment.forEach((comment) => {
+                         expect(comment).toBeInstanceOf(Object)
+                         expect(typeof comment.comment_id).toBe('number')
+                         expect(typeof comment.votes).toBe('number')
+                         expect(typeof comment.created_at).toBe('string')
+                         expect(typeof comment.author).toBe('string')
+                         expect(typeof comment.body).toBe('string')
+                     })
+                 })
+                 })
+                 })
+                 
```

Comment:
I would add another test for when the post body does not contain the required properties.